### PR TITLE
RAJA + Umpire: Add new versions

### PR DESCRIPTION
**Pull request by white238**
Should I change the master version to main now that main is a supported version?

@davidbeckingsale

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
This looks good to me @white238 . I'd like with the name of the main version to be changed if possible.

**Review by becker33 — APPROVED** [review]

**Comment by adrienbernede** [comment]
Since "examples" variant is activated by default in Umpire, I want to test this with our "spack aware" CI, for peace of mind ;)

**Comment by white238** [comment]
> Since "examples" variant is activated by default in Umpire, I want to test this with our "spack aware" CI, for peace of mind ;)

This keeps the existing functionality from Umpire which is to have examples on by default.  This just allows you to turn them off if for some reason they don't build.

**Comment by adrienbernede** [comment]
> > Since "examples" variant is activated by default in Umpire, I want to test this with our "spack aware" CI, for peace of mind ;)
> 
> This keeps the existing functionality from Umpire which is to have examples on by default. This just allows you to turn them off if for some reason they don't build.

Then it looks good to me!

File: var/spack/repos/builtin/packages/umpire/package.py [hunk 1]
```diff
@@ -15,7 +15,9 @@ class Umpire(CMakePackage, CudaPackage):
     git      = 'https://github.com/LLNL/Umpire.git'
 
     version('develop', branch='develop', submodules='True')
-    version('master', branch='main', submodules='True')
+    version('main', branch='main', submodules='True')
+    version('4.0.1', tag='v4.0.0', submodules='True')
+    version('4.0.0', tag='v4.0.0', submodules='True')
```
Comment (adrienbernede) [inline var/spack/repos/builtin/packages/umpire/package.py]: version 4.0.1 points to the same tag as version 4.0.0, is that normal or a typo?

**Review by adrienbernede — COMMENTED** [review]
I still have one question

**Review by adrienbernede — APPROVED** [review]
Good to go!

**Review by alalazo — APPROVED** [review]

**Pull request merged**